tools/generate-bolts.py

```diff
@@ -102,12 +105,12 @@ def __init__(self):
         self.optional_fields = False
         self.len_fields = {}
 
-    def add_data_field(self, field_name, type_obj, count=1, is_optional=[]):
+    def add_data_field(self, field_name, type_obj, count=1, is_optional=[], comments=[]):
```

Comment:
Having arrays as default parameters in python functions is a Bad Idea :tm:: they are allocated statically and so anything that modifies them inside the function itself will cause them to be modified for all calls:

```python

def test(arg=[]):
    arg.append(1)
    print(arg)

test()
test()
```

will result in the following output:

```sh
[1]
[1, 1]
```

It's not a big issue since we use them here in a read-only way, but something to keep in mind.

Reply:
good to know. i think i handled this correctly then for the comment sets (passing a copy of the list down instead of the list itself).